Correct references to the `zfsonlinux` org in developer docs (#61)
openzfs/openzfs-docs: docs/Developer Resources/Building ZFS.rst

GitHub Repositories
~~~~~~~~~~~~~~~~~~~

The official source for OpenZFS is maintained at GitHub by the
`openzfs <https://github.com/openzfs/>`__ organization. The primary
git repository for the project is the `zfs
<https://github.com/openzfs/zfs>`__ repository.

There are two main components in this repository:

> - **ZFS**: The ZFS repository contains a copy of the upstream OpenZFS

Reviewer: In terms of eliminating the "copy of upstream", maybe something like this:

**ZFS**: The ZFS repository contains the OpenZFS code which supports FreeBSD and Linux. The vast majority of the OpenZFS code is OS-independent.

Reviewer 2: @rlaager I agree "copy of upstream" should be gone. However:

"The vast majority of the OpenZFS code is OS-independent."

 What do you want to say here?
- The original meaning was the fact it was "self contained" and usable without modification.
- Your version doesn't seem to have the same meaning, it's rather vague.
- also "The ZFS repository contains the OpenZFS code which supports FreeBSD and Linux", is rather vague too... It could just as well be read "contains compatibility code" or "Contains code that supports FreeBSD and Linux" (without being clear it's the main repo for the project)

How about something like, to keep the original meaning in tact:
```suggestion
- **ZFS**: The ZFS repository contains the main OpenZFS code with support for FreeBSD and Linux. The vast majority of the OpenZFS code is OS-independent and can be used without modification.
```

Reviewer: In terms of "self contained" vs "OS-independent", I actually have the same question: I have no idea what the original intention was.

If the audience of this is a user of ZFS, rather than someone porting it to a new platform, I'm not sure that we need the second sentence anyway.

Reviewer 2: @rlaager You are right, I was getting confused by that second scentence., reading both versions this tells all we need:

```suggestion
- **ZFS**: The ZFS repository contains the OpenZFS code which supports FreeBSD and Linux.
```
_(or my version of the same scentence, thats more of a style choice)_

That being said:
Shouldn't there be a comma after code?
`The ZFS repository contains the OpenZFS code, which supports FreeBSD and Linux.`
vs
`The ZFS repository contains the OpenZFS code which supports FreeBSD and Linux.`

afaik one of those is:
"the code to supports something" and the other "the code having/including support for something"

> - **ZFS**: The ZFS repository contains a copy of the upstream OpenZFS
>    code which has been adapted and extended for Linux and FreeBSD. The
>    vast majority of the core OpenZFS code is self-contained and can be
>    used without modification.

Reviewer 3: ```suggestion
- **ZFS**: The OpenZFS repository contains a generic code and system-dependent
   parts for Linux and FreeBSD. The vast majority of the core OpenZFS code
   is self-contained and can be used without modification.
```

Author: Thanks, wasn't sure what exactly was correct here. Typo: "contains generic code" instead of "contains a generic code"

Reviewer 3: I like @rlaager 's wording more.

~~~~~~~~~~~~~~~~~~~

> The official source for ZFS on Linux is maintained at GitHub by the

Reviewer 3: ```suggestion
The official source for OpenZFS is maintained at GitHub by the
```

Author: Nice catch :+1: fixed another instance too in a followup commit

~~~~~~~~~~~~~~~~~~~

The official source for ZFS on Linux is maintained at GitHub by the
`openzfs <https://github.com/openzfs/>`__ organization. The project
consists of two primary git repositories named `spl
<https://github.com/openzfs/spl>`__ and `zfs
<https://github.com/openzfs/zfs>`__, both are required to build ZFS on

> Linux.

Reviewer 4: The `spl` repo hasn't been used for quite a while now, I think you can remove references to it (per the note below), and just address building 0.8 and later code.

Author: @ahrens pushed a new commit removing all references to building <= 0.7. Sorry for the ugly diff :/
